### PR TITLE
refactor: extract plugin manifest sync into schema-aware module

### LINKED ISSUE #95: release: make plugin manifest sync schema-aware and extract to module
## Summary

`scripts/release.sh` synchronizes `.claude-plugin/plugin.json` and `.claude-plugin/marketplace.json` from `package.json` by writing a known allowlist of keys and deleting a known denylist (`plugin.bugs`, `marketplace.$schema`, root-level `version`/`description`). The sync logic lives inline as two multi-line `node -e` blocks. This works against the current Claude Code plugin/marketplace schema but silently drops any new schema-required fields that appear in future CLI releases, and the inline form is difficult to lint or unit-test.

## Evidence

- `scripts/release.sh` `Syncing plugin.json` block copies a fixed set of keys and `delete plugin.bugs;`.
- `scripts/release.sh` `Syncing marketplace.json` block copies a fixed set of keys, deletes `$schema`, root `version`, and root `description`.
- `docs/project-marketplace-distribution.md` **Risks** table records both concerns: "Claude Code plugin schema gains new required fields that the sync script does not propagate" and "Inline `node -e` blocks in `release.sh` drift from the schema and become hard to maintain."
- The `validate-plugin` CI job added in #92 catches **shape** regressions but not **incompleteness** — a future schema that adds a new required field with no default would break release time, not PR time.

## Current Gap

- The sync cannot preserve unknown keys that a future schema version introduces.
- A schema evolution that adds a required field with no default would cause every `release.sh` run to fail hard with no workaround short of editing the script.
- The inline `node -e` blocks cannot be linted, cannot be unit-tested, and must be kept in sync across two similar-but-different manifest shapes.

### In Scope

- Extract both sync blocks into a standalone `scripts/release-sync-manifests.mjs` module invoked from `scripts/release.sh`.
- Replace the allowlist/denylist pattern with a schema-aware merge: preserve any existing key that the current schema does not explicitly forbid, overwrite only the keys the sync owns (version, name, description, author, license, homepage, repository).
- Add a unit test harness under `scripts/__tests__/` or `skills/design-farmer/tests/` that exercises: (a) current manifest shape, (b) a manifest with an unknown optional field (should be preserved), (c) a manifest missing a known-overwritten field (should be added).
- Keep the existing strict checks (`pkg.description` required, `mp.plugins` non-empty) and the current schema-forbidden deletes (`plugin.bugs`, `mp.$schema`, root `version`/`description`) as module constants.

### Out of Scope

- Shipping an actual JSON schema file. The current `claude plugin validate .` gate is the enforcement path.
- Runtime plugin resolution behavior (separate concern).
- Porting other inline `node -e` usages elsewhere in the repo.

## Acceptance Criteria

- [ ] `scripts/release-sync-manifests.mjs` exists, exports pure functions for `plugin.json` and `marketplace.json` synchronization, and is invoked from `scripts/release.sh`.
- [ ] Running the sync on a manifest with an unknown top-level field preserves that field.
- [ ] Unit tests cover the three scenarios listed under In Scope.
- [ ] `claude plugin validate .` still passes after the refactor.
- [ ] `docs/marketplace-release-procedure.md` and `docs/project-marketplace-distribution.md` are updated to reference the new module and reflect the schema-aware behavior.

## Out of Scope

Same as the Out of Scope subsection under Proposed Scope above.

## Additional Notes

Tracked as a follow-up from #92. Bundles two related follow-ups from that review cycle (schema-aware merge + inline-node extraction) into one atomic refactor.

### PR DESCRIPTION
**Pull request by hakilee**
## Summary

Closes #95.

Replace the two inline `node -e` blocks in `scripts/release.sh` that sync `.claude-plugin/plugin.json` and `.claude-plugin/marketplace.json` with a standalone ESM module that does a **schema-aware merge**: every existing manifest key the sync does not explicitly own survives untouched, so a future Claude Code plugin schema that adds a new optional field (or a maintainer hand-adding one) is no longer silently dropped on the next release.

The refactor also makes the sync logic testable. `scripts/release-sync-manifests.mjs` exports pure functions that can be unit-tested in isolation, and `scripts/__tests__/release-sync-manifests.test.mjs` runs 24 cases against the module under `node --test`.

Running the new sync against the current committed manifests produces **byte-identical** output, so this commit introduces no runtime change for the happy path — the new behavior only manifests when an unknown manifest field exists.

## Changes

- **`scripts/release-sync-manifests.mjs`** (new)
  - Pure library functions: `syncPluginManifest`, `syncMarketplaceManifest`, `normalizeAuthor`, `normalizeRepository`.
  - CLI entry point for `release.sh`: `node scripts/release-sync-manifests.mjs <plugin|marketplace> <package.json> <manifest.json>`.
  - Schema-aware merge preserves every existing manifest key except the explicitly-forbidden ones (`plugin.bugs`, `marketplace.$schema`, marketplace root-level `version`/`description`).
  - Required-field guards on `pkg.description` and `marketplace.plugins` non-empty.
- **`scripts/__tests__/release-sync-manifests.test.mjs`** (new) — 24 `node:test` cases:
  - Normalizer edge cases (string vs object author, repository object unwrapping, missing input)
  - Baseline manifest shape round-trips for both plugin and marketplace
  - Unknown-top-level-field preservation (plugin and marketplace)
  - Unknown-first-plugin-field preservation
  - Adding missing canonical fields
  - Promoting marketplace root-level version/description into metadata
  - Stripping `plugin.bugs` and `marketplace.$schema`
  - Input-manifest immutability (caller's object is never mutated)
  - Required-field guards
- **`scripts/release.sh`**
  - Replaced the two inline `node -e` sync blocks with CLI calls to the new module.
  - Pre-release validation now runs `node --test scripts/__tests__/release-sync-manifests.test.mjs` alongside the structural suite, semantic suite, and `claude plugin validate .`.
- **`.github/workflows/skill-quality.yml`**
  - `validate-plugin` job gained a "Run manifest sync unit tests" step before installing the Claude Code CLI.
- **`AGENTS.md`**
  - Structure map lists both the module and its test file.
  - CI Baseline section mentions the manifest sync unit test step inside `validate-plugin`.
- **`docs/marketplace-release-procedure.md`**
  - "What the Release Script Does" step 1 lists the new `node --test` gate.
  - Step 4 describes the module-based sync and highlights the schema-aware merge.
- **`docs/project-marketplace-distribution.md`**
  - Risks table: the schema-gain row is upgraded from "follow-up" to "gated by unit tests + schema-aware merge".
  - Risks table: the inline-`node -e` row is marked resolved in #95.

## Validation

```
$ node --test scripts/__tests__/release-sync-manifests.test.mjs
ℹ tests 24
ℹ suites 0
ℹ pass 24
ℹ fail 0

$ bash -n scripts/release.sh
(no errors)

$ ruby -ryaml -e "YAML.load_file('.github/workflows/skill-quality.yml')"
(no errors)

$ claude plugin validate .
✔ Validation passed

$ bash scripts/validate-skill-md.sh
All skill structure and contract checks passed.
```

**Byte-identical output confirmation** — ran the new sync module against the currently committed manifests and diffed the output against the originals:

```
--- plugin.json diff ---
no diff
--- marketplace.json diff ---
no diff
```

## Out of Scope

- Shipping an actual JSON schema file for plugin/marketplace manifests. The current `claude plugin validate .` gate is the enforcement path.
- Runtime plugin resolution behavior (separate concern tracked in #93).
- Porting other inline `node -e` usages elsewhere in the repo.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by hakilee** [comment]
@codex review

File: scripts/release-sync-manifests.mjs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  const author = normalizeAuthor(pkg.author);
+  if (author) {
+    next.author = author;
+  }
+
+  if (pkg.license) {
+    next.license = pkg.license;
+  }
+  if (pkg.homepage) {
+    next.homepage = pkg.homepage;
```
[the remark this answers is not in the document]
Comment (hakilee) [reply scripts/release-sync-manifests.mjs]: **Applied.** You're right — my refactor made `license`/`homepage`/`author`/`repository` conditional, which leaves stale manifest values when those fields are removed from `package.json`. The inline logic used unconditional assignment so undefined values were JSON-dropped and effectively cleared, and I missed that on the refactor.

Fixed by introducing an `assignOrClear` helper that writes the value when present in `package.json` and explicitly `delete`s the manifest key when absent. It applies to all four canonical optional fields in `syncPluginManifest` and to `plugins[0].homepage` in `syncMarketplaceManifest`. Unknown (non-canonical) keys like `skills` or future optional fields are still preserved verbatim because the helper only acts on fields the sync explicitly owns.

Added three new unit tests covering the cleared-field behavior and verifying that clearing canonical fields does not touch unknown keys (27 tests total, all passing). Byte-identical output against the current committed manifests is preserved since both files still carry all four optional fields.

**Comment by hakilee** [comment]
@codex review
